snackpub/src/RedisAdapter.ts

```diff
@@ -0,0 +1,45 @@
+import { createAdapter } from '@socket.io/redis-adapter';
+import fs from 'fs';
+import { createClient } from 'redis';
+import type { Server } from 'socket.io';
+
+import Env from './Env';
+
+let pubClient: ReturnType<typeof createClient> | null = null;
+let subClient: ReturnType<typeof createClient> | null = null;
+
+export async function maybeBindRedisAdapterAsync(
+  server: Server,
+  inputOptions: Parameters<typeof createClient>[0] = {}
+) {
+  if (Env.redisURL) {
+    try {
+      const options: Parameters<typeof createClient>[0] = {
+        url: Env.redisURL,
+      };
+      if (Env.redisURL.startsWith('rediss:') && Env.redisTlsCa) {
+        options.socket = {
+          tls: true,
+          ca: fs.readFileSync(Env.redisTlsCa),
+        };
+      }
```

Comment:
If the URL starts with rediss: but there is no CA certificate, does the connection to Redis fail? (We do not want to accidentally connect to Redis without encryption, so failing is good.)

Reply:
Won't it not start with that in development?

Reply:
If we connect to a local Redis instance, I'd expect to use `redis:` (not `rediss:`) and not need a TLS certificate

Reply:
> If the URL starts with rediss: but there is no CA certificate, does the connection to Redis fail? (We do not want to accidentally connect to Redis without encryption, so failing is good.)

from my testing for the connection to redislab, it will throw errors.

> Won't it not start with that in development?

by default we don't have redis for snackpub in development, the socketio is running upon in-memory mode. if we have redis running locally with `REDIS_URL` environment variable, then we can test redis mode as well. and yes, `REDIS_URL='redis://localhost:6379'`